cmd/zc_filter.go

```diff
@@ -59,6 +59,39 @@ func (f *excludeBlobTypeFilter) DoesPass(object StoredObject) bool {
 	return false
 }
 
+// excludeContainerFilter filters out container names that must be excluded
+type excludeContainerFilter struct {
+	containerNamesList []string
+}
+
+func (s *excludeContainerFilter) DoesSupportThisOS() (msg string, supported bool) {
+	return "", true
+}
+
+func (s *excludeContainerFilter) AppliesOnlyToFiles() bool {
+	return false // excludeContainerFilter is related to container names, not related to files
+}
+
+func (s *excludeContainerFilter) DoesPass(storedObject StoredObject) bool {
+	if len(s.containerNamesList) == 0 {
+		return true
+	}
+	for _, cName := range s.containerNamesList {
+		// container name matches the excluded container name and return false to exclude
+		if strings.Compare(cName, storedObject.ContainerName) == 0 {
```

Comment:
If its an exaxt match that we are doing then it makes more sense to store the configured list in a map and then do a search. Iterating serially and comparing will work fine for smaller list but if someone gives a large number of containers to be excluded this will hit the performance and CPU usage.

Reply:
Great idea and now updated!